pandas/core/frame.py

```diff
@@ -2239,9 +2241,10 @@ def to_feather(self, path, **kwargs) -> None:
     )
     def to_markdown(
         self,
-        buf: Optional[IO[str]] = None,
-        mode: Optional[str] = None,
+        buf: Optional[Union[IO[str], str]] = None,
+        mode: Optional[str] = "wt",
```

Comment:
Why did this need to change?

Reply:
mode=None doesn't actually make sense. This is for writing, and most open commands (including python builtin) would default to "rt" or "rb".

Reply:
if changing this, could also remove Optional from type annotation.